src/core/metadata_manager.py

```diff
@@ -39,27 +39,26 @@
 from lib.timing import get_utc_now, global_start
 
 
-@Singleton
 class MetadataManager:
-    metadata_extractors: list[type(MetadataBase)] = []
-
-    blacklisted_for_cache = [MaliciousExtensions, ExtractFromFiles, Javascript]
-
     def __init__(self) -> None:
         self._logger = get_logger()
-        self._setup_extractors()
+        self.metadata_extractors: list[MetadataBase] = [
+            self._setup_extractor(extractor) for extractor in self._extractors()
+        ]
         self.tld_extractor: TLDExtract = TLDExtract(cache_dir=None)
+        self.blacklisted_for_cache = [MaliciousExtensions, ExtractFromFiles, Javascript]
+        self.cache_manager = CacheManager()
 
     def _setup_extractor(self, extractor_class: type(MetadataBase)) -> MetadataBase:
         self._logger.debug(f"Starting setup for {extractor_class} {get_utc_now()}")
         extractor = extractor_class(self._logger)
-        extractor.setup()
+        asyncio.run(extractor.setup())
```

Comment:
This was actually a bug! Some of the extactors were never setup correctly and hence operated on empty tag lists :D

Reply:
Good you found that. Should one of the tests have noticed his?

Reply:
The manager test could have caught it. I think to test the manager the way to go is injecting a mock extractor and testing that it was setup and its mocked result is part of the manager output.